lib/computer_player.rb

```diff
@@ -1,17 +1,80 @@
+require_relative 'win_checker'
+require_relative 'score_tracker'
+
 class ComputerPlayer < Player
 
-  def initialize(player_num, board)
-    @board = board
+  def initialize(player_num, side_length)
+    @player_num = player_num
+    @next_move = 0
+    @side_length = side_length
+    @comp_token = "O"
+    @other_token = "X"
   end
 
-  def get_next_move
-    available_moves = @board.get_available_moves
+  def get_next_move(board)
+    minimax_score(board.to_string_array, @player_num)
 
-    available_moves[0].to_s
+    @next_move.to_s
   end
 
   def get_token
-    "O"
+    @comp_token
+  end
+
+  private
+
+  def minimax_score(board_string_array, next_player)
```

Comment:
Why pass a `board_string_array`?  Why not pass a `Board`?  Converting back and forth between `Board`s and `board_string_array`s seems like your way to clone a `Board`.  Wouldn't it be cleaner to ask the `Board` for a clone of itself?  Or, better yet, make your `Board` immutable.  Whenever the `Board` is altered, return a new `Board` with the new state.

Reply:
Passing Board directly into minimax_score now.